src/battle_game/core/units/tiny_tank.cpp

```diff
@@ -86,25 +89,42 @@ void Tank::Update() {
 }
 
 void Tank::TankMove(float move_speed, float rotate_angular_speed) {
+  // rand(time(0));
   auto player = game_core_->GetPlayer(player_id_);
   if (player) {
     auto &input_data = player->GetInputData();
     glm::vec2 offset{0.0f};
+    bool fx = 0;
     if (input_data.key_down[GLFW_KEY_W]) {
       offset.y += 1.0f;
+      fx = 1;
     }
     if (input_data.key_down[GLFW_KEY_S]) {
       offset.y -= 1.0f;
+      fx = 1;
     }
     float speed = move_speed * GetSpeedScale();
     offset *= kSecondPerTick * speed;
     auto new_position =
         position_ + glm::vec2{glm::rotate(glm::mat4{1.0f}, rotation_,
                                           glm::vec3{0.0f, 0.0f, 1.0f}) *
                               glm::vec4{offset, 0.0f, 0.0f}};
+    // printf("Before:%f %f\n",new_position.x,new_position.y);
+    if (fx) {
+      glm::vec2 offset2{0.0f};
+      // printf("%d\n",rand());
+      offset2.y += (rand() - (1 << 30)) / 1000000000.0;
+      offset2 *= kSecondPerTick * speed;

```

Comment:
Why are you using rand() function? I gave you RandomFloat() in GameCore and told you in docs. Undo this!
BTW, **DO NOT** make changes in the example code.